src/scene.ts

```diff
@@ -4481,7 +4481,11 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
         this.metadata = null;
 
         if (EngineStore._LastCreatedScene === this) {
-            EngineStore._LastCreatedScene = null;
+            if (this._engine.scenes.length > 1) {

```

Comment:
Wouldn't this entire code fit better after removing the scene from `engine.scenes` in line 485?